connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java

```diff
@@ -236,20 +239,28 @@ protected void poll(long timeoutMs) {
         }
     }
 
+    private void doCommitSync(Map<TopicPartition, OffsetAndMetadata> offsets, int seqno) {
+        try {
+            consumer.commitSync(offsets);
+            lastCommittedOffsets = offsets;
+            onCommitCompleted(null, seqno);
+        } catch (WakeupException e) {
+            // retry the commit to ensure offsets get pushed, then propagate the wakeup up to poll
+            doCommitSync(offsets, seqno);
+            throw e;
```

Comment:
Hmm... I think the intent was to propagate the wakeup to poll() because the task is closing, but actually it looks like this would propagate up to execute() instead. I think we need to move the `catch` in poll() to iteration() instead. I'll submit a patch.